DigitalLearningSolutions.Web/Controllers/TrackingSystem/CentreConfiguration/RegistrationPromptsController.cs

```diff
@@ -142,5 +148,59 @@ private void IgnoreAddNewAnswerValidation()
                 ModelState[key].ValidationState = ModelValidationState.Valid;
             }
         }
+
+        [HttpGet]
+        [Route("Add/SelectPrompt")]
+        public IActionResult AddRegistrationPromptSelectPrompt()
+        {
+            SetViewBagCustomPromptNameOptions();
+
+            var addRegistrationPromptData = TempData.Peek<AddRegistrationPromptData>();
+
+            if (addRegistrationPromptData == null || !Request.Cookies.ContainsKey(CookieName))
+            {
+                addRegistrationPromptData = new AddRegistrationPromptData();
+                var id = addRegistrationPromptData.Id;
+
+                Response.Cookies.Append(
+                    CookieName,
+                    id.ToString(),
+                    new CookieOptions
+                    {
+                        Expires = DateTimeOffset.UtcNow.AddDays(1)
+                    });
+                TempData.Set(addRegistrationPromptData);
+
+                return View(addRegistrationPromptData.SelectPromptViewModel);
+            }
+
+            return View(addRegistrationPromptData.SelectPromptViewModel);
+        }
+
+        [HttpPost]
+        [Route("Add/SelectPrompt")]
+        [ServiceFilter(typeof(RedirectEmptySessionData<AddRegistrationPromptData>))]
+        public IActionResult AddRegistrationPromptSelectPrompt(AddRegistrationPromptSelectPromptViewModel model)
+        {
+            if (!ModelState.IsValid)
+            {
+                SetViewBagCustomPromptNameOptions();
+                return View(model);
+            }
+
+            var data = TempData.Peek<AddRegistrationPromptData>()!;
+            data.SelectPromptViewModel = model;
+            TempData.Set(data);
+
+            // will redirect to next page in multi page form

```

Comment:
We usually add comments with a TODO and a JIRA reference:
`// TODO: HEEDLS-452 - redirect to next page in multi-page form `